File: test/resources/HelloProphecy/prophecy_scala/pipelines/silver_customers_orders/code/src/main/scala/io/prophecy/pipelines/silver_customers_orders/graph/UniqueZipCodes.scala
package io.prophecy.pipelines.silver_customers_orders.graph

import io.prophecy.libs._
import io.prophecy.pipelines.silver_customers_orders.udfs.PipelineInitCode._
import io.prophecy.pipelines.silver_customers_orders.udfs.UDFs._
import io.prophecy.pipelines.silver_customers_orders.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object UniqueZipCodes {

  def apply(context: Context, in0: DataFrame): DataFrame = {
    in0.createOrReplaceTempView("in0")
    context.spark.sql(
      """SELECT 
    row_number() OVER (PARTITION BY 1 ORDER BY zipcode ASC) AS row_number, 
    zipcode 
FROM (
    SELECT distinct zipcode from in0 order by zipcode asc
)"""
    )
  }

}

File: test/resources/HelloProphecy/prophecy/pipelines/silver_customers_orders/code/silver_customers_orders/graph/ZipCodes/UniqueZipCodes.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from .config import *
from silver_customers_orders.udfs.UDFs import *

def UniqueZipCodes(spark: SparkSession, in0: DataFrame) -> (DataFrame):

    try:
        registerUDFs(spark)
    except NameError:
        print("registerUDFs not working")

    in0.createOrReplaceTempView("in0")
    df1 = spark.sql(
        "SELECT \n    row_number() OVER (PARTITION BY 1 ORDER BY zipcode ASC) AS row_number, \n    zipcode \nFROM (\n    SELECT distinct zipcode from in0 order by zipcode asc\n)"
    )

    return df1
